diff -Naur -x '*~' dvdisaster-0.72.1/memtrack.c dvdisaster-0.72.1-fix-format-errors/memtrack.c
--- dvdisaster-0.72.1/memtrack.c	2009-07-06 20:32:15.000000000 +0200
+++ dvdisaster-0.72.1-fix-format-errors/memtrack.c	2009-10-07 21:24:52.000000000 +0200
@@ -169,7 +169,7 @@
 static void print_ptrs(char *msg)
 {  int bucket,i,n=0;
 
-   g_printf(msg);
+   g_printf("%s", msg);
 
    for(bucket=0; bucket<64; bucket++)
       for(i=0; i<phCnt[bucket]; i++)
diff -Naur -x '*~' dvdisaster-0.72.1/misc.c dvdisaster-0.72.1-fix-format-errors/misc.c
--- dvdisaster-0.72.1/misc.c	2009-07-06 20:32:15.000000000 +0200
+++ dvdisaster-0.72.1-fix-format-errors/misc.c	2009-10-07 21:28:23.000000000 +0200
@@ -208,7 +208,7 @@
       legislature. If in doubt, just translate the following like "This is free
       software; please refer to the conditions of the GNU GENERAL PUBLIC LICENSE
       in the source code." Avoid making any legal statements by your own.*/
-   g_fprintf(where, _("This software comes with  ABSOLUTELY NO WARRANTY.  This\n"
+   g_fprintf(where, "%s", _("This software comes with  ABSOLUTELY NO WARRANTY.  This\n"
 		      "is free software and you are welcome to redistribute it\n"
 		      "under the conditions of the GNU GENERAL PUBLIC LICENSE.\n"  
 		      "See the file \"COPYING\" for further information.\n"));
@@ -374,7 +374,7 @@
       log_window_append(tmp2);
    }
    else
-   {  g_fprintf(stderr, tmp2);
+   {  g_fprintf(stderr, "%s", tmp2);
 
       fflush(stderr);   /* at least needed for Windows */
    }
@@ -498,7 +498,7 @@
    }
 
    if(!Closure->guiMode) 
-   {  g_fprintf(stderr, _("\n*\n* dvdisaster - can not continue:\n*\n"));
+   {  g_fprintf(stderr, "%s", _("\n*\n* dvdisaster - can not continue:\n*\n"));
       va_start(argp, format);
       g_vfprintf(stderr, format, argp);
       va_end(argp);
@@ -830,6 +830,7 @@
 				   GTK_DIALOG_DESTROY_WITH_PARENT,
 				   mi->message_type,
 				   mi->button_type,
+                   "%s",
 				   mi->msg);
    gtk_label_set_line_wrap(GTK_LABEL(((struct _GtkMessageDialog*)dialog)->label), FALSE);
 
diff -Naur -x '*~' dvdisaster-0.72.1/preferences.c dvdisaster-0.72.1-fix-format-errors/preferences.c
--- dvdisaster-0.72.1/preferences.c	2009-07-06 20:32:25.000000000 +0200
+++ dvdisaster-0.72.1-fix-format-errors/preferences.c	2009-10-07 21:26:29.000000000 +0200
@@ -1323,6 +1323,7 @@
 						    GTK_DIALOG_DESTROY_WITH_PARENT,
 						    GTK_MESSAGE_QUESTION,
 						    GTK_BUTTONS_OK_CANCEL,
+                            "%s",
 						    _utf("Delete the log file?"));
 	 int answer;
 	   
diff -Naur -x '*~' dvdisaster-0.72.1/read-adaptive.c dvdisaster-0.72.1-fix-format-errors/read-adaptive.c
--- dvdisaster-0.72.1/read-adaptive.c	2009-07-06 20:32:15.000000000 +0200
+++ dvdisaster-0.72.1-fix-format-errors/read-adaptive.c	2009-10-07 21:30:41.000000000 +0200
@@ -1029,7 +1029,7 @@
      {  int seq = (j/2000)%10;
 
 	if(!Closure->guiMode)
-        {  g_printf(anim[seq]);
+        {  g_printf("%s", anim[seq]);
 	   fflush(stdout);   /* at least needed for Windows */
 	}
      }
diff -Naur -x '*~' dvdisaster-0.72.1/read-linear.c dvdisaster-0.72.1-fix-format-errors/read-linear.c
--- dvdisaster-0.72.1/read-linear.c	2009-07-06 20:32:25.000000000 +0200
+++ dvdisaster-0.72.1-fix-format-errors/read-linear.c	2009-10-07 21:32:52.000000000 +0200
@@ -1339,19 +1339,19 @@
 	       else if(   rc->readOK == rc->sectors  /* no user limited range */  
 		       && rc->pass == 1              /* md5sum invalid after first pass */
 		       && memcmp(rc->ei->eh->mediumSum, Closure->md5Cache, 16))
-		  t = g_strdup_printf(_("All sectors successfully read, but wrong image checksum."));
-	       else t = g_strdup_printf(_("All sectors successfully read. Checksums match."));
+		  t = g_strdup_printf("%s", _("All sectors successfully read, but wrong image checksum."));
+	       else t = g_strdup_printf("%s", _("All sectors successfully read. Checksums match."));
 	       break;
 	    case ECC_RS02:
 	       if(!md5_failure)
-		    t = g_strdup_printf(_("All sectors successfully read. Checksums match."));
+		    t = g_strdup_printf("%s", _("All sectors successfully read. Checksums match."));
 	       else
 	       {    t = g_strdup_printf(_("All sectors successfully read, %s!"), md5_failure);
 		    g_free(md5_failure); md5_failure=NULL;
 	       }
 	       break;
 	    default:
-	       t = g_strdup_printf(_("All sectors successfully read."));
+	       t = g_strdup_printf("%s", _("All sectors successfully read."));
 	       break;
 	 }
       }
diff -Naur -x '*~' dvdisaster-0.72.1/scsi-layer.c dvdisaster-0.72.1-fix-format-errors/scsi-layer.c
--- dvdisaster-0.72.1/scsi-layer.c	2009-08-06 12:12:21.000000000 +0200
+++ dvdisaster-0.72.1-fix-format-errors/scsi-layer.c	2009-10-07 21:33:48.000000000 +0200
@@ -1871,7 +1871,7 @@
 			"READ DVD STRUCTURE: %lld sectors\n\n"),
 		      dh->readCapacity+1, dh->userAreaSize+1);
 
-      g_string_append_printf(warning, _("Evaluation of returned medium sizes:\n\n"));
+      g_string_append_printf(warning, "%s", _("Evaluation of returned medium sizes:\n\n"));
 
       /*** Look at READ CAPACITY results */
 
diff -Naur -x '*~' dvdisaster-0.72.1/tools/memory.c dvdisaster-0.72.1-fix-format-errors/tools/memory.c
--- dvdisaster-0.72.1/tools/memory.c	2009-07-06 20:32:25.000000000 +0200
+++ dvdisaster-0.72.1-fix-format-errors/tools/memory.c	2009-10-07 21:58:03.000000000 +0200
@@ -165,7 +165,7 @@
 static void print_ptrs(char *msg)
 {  int bucket,i,n=0;
 
-   fprintf(stdout, msg);
+   fprintf(stdout, "%s", msg);
 
    for(bucket=0; bucket<64; bucket++)
       for(i=0; i<phCnt[bucket]; i++)
